conda-incubator/conda-store
# Create catch-all/fallback route for UI app

Fixes #899.

## Pull request checklist



- n/a Did you update the documentation (if required)?


## How to test

Before starting, create an environment named "test" under the "default" namespace.

Set `url_prefix` to "/":

- localhost:8080/ -> 200, should load app
- localhost:8080/foo -> 200, should see 404 page (but HTTP status code should be 200)
- localhost:8080/foo/bar -> 200, UI loads with sidebar and blank main panel (this needs to be handled better by the client, the client thinks this is a route for namespace/environment, tries to request namespace=foo, environment=bar, but currently doesn't handle the error case, so it's basically a silent error)
- localhost:8080/default/test -> 200, should load the app page for the default/test environment
- localhost:8080/not-found -> 404 with page generated from React app, not the server. How can you tell? The server returns JSON.

Set `url_prefix` to "/conda-store", go to the following URLs, behavior should be exactly the same as above:

- localhost:8080/conda-store 
- localhost:8080/conda-store/foo
- localhost:8080/conda-store/foo/bar
- localhost:8080/conda-store/default/test
- localhost:8080/conda-store/not-found
